### PR TITLE
feat: add screen upload document (CMC-1490) (1/2)

### PR DESCRIPTION
**Pull request by devmarcy**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/CMC-1490

### Change description ###
- created a new YesOrNo screen which will either go to upload documents or skip that and go to claim value page
- created `uploadParticularsOfClaimQuestion` e2e page
- removed `assertError` for asserting `'You must add Particulars of claim details'` in AddOrAmendClaimDocuments event (api), as now the particulars of claim doc and text is optional

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: e2e/pages/createClaim/uploadParticularsOfClaimQuestion.page.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        yes: 'Yes',
+        no: 'No'
+      }
+    },
+  },
+
+  async chooseIfUploadParticularsOfClaim(uploadParticularsOfClaimOption) {
+    // eslint-disable-next-line no-prototype-builtins
+    if (!this.fields.uploadQuestion.options.hasOwnProperty(uploadParticularsOfClaimOption)) {
+      throw new Error(`Respondent represented option: ${uploadParticularsOfClaimOption} does not exist`);
```
Comment (swalker125) [inline e2e/pages/createClaim/uploadParticularsOfClaimQuestion.page.js]: copy and paste, please update. Although not sure this level of throwing an exception etc is required. Just have a choose yes / choose no function perhaps? Only the yes function will probably be used in e2e

**Review by sabahirfan — DISMISSED** [review]

**Review by sabahirfan — DISMISSED** [review]

**Review by dharmendrak — DISMISSED** [review]

**Review by sabahirfan — DISMISSED** [review]

**Review by dharmendrak — DISMISSED** [review]

**Review by dharmendrak — DISMISSED** [review]

File: ccd-definition/CaseEventToComplexTypes/AddOrAmendClaimDocuments.json [hunk 2]
```diff
@@ -52,9 +52,18 @@
...
+    "ID": "Upload",
+    "CaseEventID": "CREATE_CLAIM",
+    "CaseFieldID": "servedDocumentFiles",
+    "ListElementCode": "medicalReport",
     "DisplayContext": "OPTIONAL",
-    "Min": 1,
-    "FieldShowCondition": "claimType=\"CLINICAL_NEGLIGENCE\" OR claimType=\"PERSONAL_INJURY\""
+    "FieldDisplayOrder": 6,
+    "FieldShowCondition": "claimType=\"CLINICAL_NEGLIGENCE\" OR claimType=\"PERSONAL_INJURY\"",
+    "Min": 1
```
Comment (swalker125) [inline ccd-definition/CaseEventToComplexTypes/AddOrAmendClaimDocuments.json]: this seems suspect, I don't understand why we have to define medical reports again.

**Review by swalker125 — DISMISSED** [review]

**Review by dharmendrak — DISMISSED** [review]

File: build.gradle [hunk 3]
```diff
@@ -280,6 +280,9 @@ dependencies {
   implementation group: 'org.springframework.security', name: 'spring-security-core', version: '5.4.6'
   implementation group: 'org.springframework.security', name: 'spring-security-web'
   implementation group: 'org.springframework.security', name: 'spring-security-config'
+
+  // CVE-2021-22112 - Privilege Escalation
+  implementation group: 'org.springframework.security', name: 'spring-security-core', version: '5.4.6'
```
Comment (sabahirfan) [inline build.gradle]: I don't think we need spring security for this repo at all. we can simply remove these unused dependencies.

**Review by dharmendrak — APPROVED** [review]

File: ccd-definition/CaseEventToComplexTypes/AddOrAmendClaimDocuments.json [hunk 4]
```diff
@@ -52,9 +52,18 @@
...
+  {
+    "LiveFrom": "01/01/2017",
+    "ID": "Upload",
+    "CaseEventID": "CREATE_CLAIM",
+    "CaseFieldID": "servedDocumentFiles",
+    "ListElementCode": "medicalReport",
     "DisplayContext": "OPTIONAL",
-    "Min": 1,
-    "FieldShowCondition": "claimType=\"CLINICAL_NEGLIGENCE\" OR claimType=\"PERSONAL_INJURY\""
+    "FieldDisplayOrder": 6,
```
Comment (sabahirfan) [inline ccd-definition/CaseEventToComplexTypes/AddOrAmendClaimDocuments.json:64]: is it 6 or 7 ?

Comment (devmarcy) [reply ccd-definition/CaseEventToComplexTypes/AddOrAmendClaimDocuments.json:64]: it is the same as ```medicalReport.document``` so should be 6 in this case

**Review by sabahirfan — APPROVED** [review]

**Review by pd-dudkiewicz — APPROVED** [review]

**Pull request merged**